packages/dm-core-plugins/src/blueprint/BlueprintPlugin.tsx

```diff
@@ -1,231 +1,28 @@
-            <option key={attribute.attributeType}>
-              {attribute.attributeType}
-            </option>
-          </Select>
-        </div>
-        {!['string', 'number', 'boolean'].includes(attribute.attributeType) && (
-          <div style={{ marginLeft: '10px' }}>
-            <BlueprintPicker
-              label={'Select'}
-              onChange={(selectedBlueprint: string) =>
-                setAttribute({
-                  ...attribute,
-                  attributeType: selectedBlueprint,
-                })
-              }
-              type={'button'}
-            />
-          </div>
-        )}
-      </div>
-      <Spacer />
-      <Label htmlFor='default' label='Default value' />
-      {attribute.attributeType === 'boolean' ? (
-        <>
-          <Switch
-            defaultChecked={!!attribute?.default ?? false}
-            onChange={(e: any) =>
-              setAttribute({ ...attribute, default: e.target.checked })
-            }
-          />
-          {attribute.default ? String(attribute.default) : 'false'}
-        </>
-      ) : (
-        <Input
-          type={attribute.attributeType === 'number' ? 'number' : 'text'}
-          id='default'
-          value={attribute?.default || ''}
-          onChange={(event: ChangeEvent<HTMLInputElement>) => {
-            let value: string | number = event.target.value
-            if (attribute.attributeType == 'number') value = Number(value)
-            setAttribute({ ...attribute, default: value })
-          }}
-          style={{ width: INPUT_FIELD_WIDTH }}
-        />
-      )}
-      <Spacer />
-      <div style={{ display: 'flex', alignItems: 'flex-end' }}>
-        <TextField
-          id='dimensions'
-          label={'Dimensions'}
-          value={attribute?.dimensions || ''}
-          onChange={(event: ChangeEvent<HTMLInputElement>) =>
-            setAttribute({ ...attribute, dimensions: event.target.value })
-          }
-          style={{ width: '140px' }}
-        />
-        <Label label='Example: "*,2,99"' />
-      </div>
-      {!Object.values(EPrimitiveTypes).includes(attribute.attributeType) && (
-        <Switch
-          label='Contained'
-          defaultChecked={attribute?.contained ?? true}
-          onChange={(e: any) =>
-            setAttribute({ ...attribute, contained: e.target.checked })
-          }
-        />
-      )}
-      <Switch
-        label='Optional'
-        defaultChecked={attribute?.optional ?? false}
-        onChange={(e: any) =>
-          setAttribute({ ...attribute, optional: e.target.checked })
-        }
-      />
-    </div>
-  )
-}
-
 export const BlueprintPlugin = (props: IUIPlugin) => {
   const { idReference } = props
   const { document, isLoading, updateDocument, error } =
     useDocument<TBlueprint>(idReference)
   const [formData, setFormData] = useState<any>({ ...document }) //TODO remove any type (requires TBlueprint to be updated)
+  const dirtyState = !isEqual(document, formData)
```

Comment:
Would a "useMemo" help on performance here?

Reply:
yes, very much! Thank you for the pointer.